lib/dotcom/system_status/groups.ex

```diff
@@ -0,0 +1,337 @@
+defmodule Dotcom.SystemStatus.Groups do
+  @moduledoc """
+  A module that groups alerts into statuses for the system status widget.
+  """
+
+  alias Alerts.Alert
+
+  # &groups/2 returns an ordered data structure, sorted in the order
+  # given by `@lines`.
+  @lines ["Blue", "Orange", "Red", "Green"]
+  @green_line_branches ["Green-B", "Green-C", "Green-D", "Green-E"]
+  @routes ["Blue", "Mattapan", "Orange", "Red"] ++ @green_line_branches
+
+  def groups(alerts, time) do
+    grouped_alerts = Map.new(@routes, &{&1, alerts_for_line(alerts, &1)})
+
+    @routes
+    |> Enum.map(fn route_id ->
+      statuses =
+        grouped_alerts
+        |> Map.get(route_id)
+        |> alerts_to_statuses(time)
+        |> consolidate_duplicate_descriptions()
+        |> sort_statuses()
+        |> stringify_times()
+        |> maybe_add_now_text()
+
+      %{route_id: route_id, sub_routes: [], statuses: statuses}
+    end)
+    |> combine_green_line_branches()
+    |> combine_mattapan_with_red_line()
+    |> sort_routes_and_sub_routes()
+  end
+
+  # Given `alerts` and `line_id`, filters out only the alerts
+  # applicable to the given line, using the alert's "informed
+  # entities".
+  defp alerts_for_line(alerts, line_id) do
+    alerts
+    |> Enum.filter(fn %Alert{informed_entity: informed_entity} ->
+      informed_entity
+      |> Enum.any?(fn
+        %{route: ^line_id} -> true
```

Comment:
confusing: Wait so does this line _not_ mean GTFS line then?

Reply:
Ugh yeah - I'm not sure where to draw the line between `route`/`line`/`branch`, etc. This probably isn't it though.